### ISSUE TITLE
np.insert gives UnitConversionError with Numpy 1.17.0 (used to work)

### ISSUE THREAD

**Issue by pllim** [astropy/astropy#9141]
Here is a minimal reproducible code snippet:
```python
import numpy as np
from astropy import units as u
PHOTLAM = u.ph / u.cm / u.cm / u.s / u.AA
flux = np.arange(5) * PHOTLAM
newflux = np.insert(flux, 0, 0)
```

With `astropy 4.0.dev25566` and `numpy 1.17.0`, I get this traceback:
```
---------------------------------------------------------------------------
UnitConversionError                       Traceback (most recent call last)
.../astropy/units/quantity.py in to_value(self, unit, equivalencies)
    716             try:
--> 717                 scale = self.unit._to(unit)
    718             except Exception:

.../astropy/units/core.py in _to(self, other)
    952         raise UnitConversionError(
--> 953             f"'{self!r}' is not a scaled version of '{other!r}'")
    954 

UnitConversionError: 'Unit(dimensionless)' is not a scaled version of 'Unit("ph / (Angstrom cm2 s)")'

During handling of the above exception, another exception occurred:

UnitConversionError                       Traceback (most recent call last)
<ipython-input-5-7e3bcc47515f> in <module>
----> 1 newflux = np.insert(flux, 0, 0)

<__array_function__ internals> in insert(*args, **kwargs)

.../astropy/units/quantity.py in __array_function__(self, function, types, args, kwargs)
   1505             function_helper = FUNCTION_HELPERS[function]
   1506             try:
-> 1507                 args, kwargs, unit, out = function_helper(*args, **kwargs)
   1508             except NotImplementedError:
   1509                 return self._not_implemented_or_raise(function, types)

.../astropy/units/quantity_helper/function_helpers.py in insert(arr, obj, values, *args, **kwargs)
    449         raise NotImplementedError
    450 
--> 451     (arr, values), unit = _quantities2arrays(arr, values)
    452     return (arr, obj, values) + args, kwargs, unit, None
    453 

.../astropy/units/quantity_helper/function_helpers.py in _quantities2arrays(*args)
    319     unit = qs[0].unit
    320     # Allow any units error to be raised.
--> 321     arrays = tuple(q.to_value(unit) for q in qs)
    322     return arrays, unit
    323 

.../astropy/units/quantity_helper/function_helpers.py in <genexpr>(.0)
    319     unit = qs[0].unit
    320     # Allow any units error to be raised.
--> 321     arrays = tuple(q.to_value(unit) for q in qs)
    322     return arrays, unit
    323 

.../astropy/units/quantity.py in to_value(self, unit, equivalencies)
    718             except Exception:
    719                 # Short-cut failed; try default (maybe equivalencies help).
--> 720                 value = self._to_value(unit, equivalencies)
    721             else:
    722                 value = self.view(np.ndarray)

.../astropy/units/quantity.py in _to_value(self, unit, equivalencies)
    649             equivalencies = self._equivalencies
    650         return self.unit.to(unit, self.view(np.ndarray),
--> 651                             equivalencies=equivalencies)
    652 
    653     def to(self, unit, equivalencies=[]):

.../astropy/units/core.py in to(self, other, value, equivalencies)
    987             return UNITY
    988         else:
--> 989             return self._get_converter(other, equivalencies=equivalencies)(value)
    990 
    991     def in_units(self, other, value=1.0, equivalencies=[]):

.../astropy/units/core.py in _get_converter(self, other, equivalencies)
    918                             pass
    919 
--> 920             raise exc
    921 
    922     def _to(self, other):

.../astropy/units/core.py in _get_converter(self, other, equivalencies)
    904         try:
    905             return self._apply_equivalencies(
--> 906                 self, other, self._normalize_equivalencies(equivalencies))
    907         except UnitsError as exc:
    908             # Last hope: maybe other knows how to do it?

.../astropy/units/core.py in _apply_equivalencies(self, unit, other, equivalencies)
    888         raise UnitConversionError(
    889             "{} and {} are not convertible".format(
--> 890                 unit_str, other_str))
    891 
    892     def _get_converter(self, other, equivalencies=[]):

UnitConversionError: '' (dimensionless) and 'ph / (Angstrom cm2 s)' (photon flux density wav) are not convertible
```

With `astropy 4.0.dev25566` and `numpy 1.16.4`, I get this:
```python
>>> newflux  # Another zero inserted at the beginning, as expected
<Quantity [0., 0., 1., 2., 3., 4.] ph / (Angstrom cm2 s)>
```

If I downgrade to `astropy 3.2.1`, it works with both versions of Numpy. I suspect this is a side effect of #8808 . Any advice, @mhvk ?

**UPDATE:** Workaround is to call `np.insert(flux, 0, 0 * flux.unit)` instead.

**Comment by pllim** [comment]
If this is a feature and not a bug, feel free to close. I found a workaround and updated the issue above with it. FYI.

**Comment by mhvk** [comment]
Hmm, the error you get is because the unit is not consistent.  In principle, like for equality tests or setting of an array item, in your specific example of setting to `0` should be allowed, I think, which means this is indeed a bug (though your PR to synphot seems a good idea - explicit is better).

**Comment by mhvk** [comment]
Changed it back to "bug", though arguably of low priority.

**Comment by mhvk** [comment]
Decided to fix this one - see #9383
